### ISSUE TITLE
Different in Database Index (Vita Version 2.1.1)

### ISSUE THREAD

**Issue by Michaellok1992** [rivantsov/vita#69]
Hi @rivantsov , i found out that there is some different in **v1.9.x** and **v2.1.1**, there is some missing index.

Common version:
![image](https://user-images.githubusercontent.com/40017655/50045374-b4504a80-00cc-11e9-8481-f4ef1085cd0e.png)

**v1.9.x**:
![image](https://user-images.githubusercontent.com/40017655/50045378-d2b64600-00cc-11e9-959c-e12a918c6717.png)

![image](https://user-images.githubusercontent.com/40017655/50045385-dfd33500-00cc-11e9-9184-d6679ae64980.png)

**v.2.1.1**:
![image](https://user-images.githubusercontent.com/40017655/50045390-03967b00-00cd-11e9-9e7f-551967de8fd8.png)

![image](https://user-images.githubusercontent.com/40017655/50045422-612ac780-00cd-11e9-857f-c31c8dca7bed.png)

---------------------------------------

**Comment by rivantsov** [comment]
Ah, OK, I see. I probably should mention it in docs in the future. This is kinda expected, change for good I think. Here is the story.
SQL Server does not index foreign key columns automatically. So you might have FK constraint, but without index. In many cases you do need index there, so in VITA 1.x the index was added automatically (if there was a flag set in DbOptions)
However, it turns out it might result in over-indexing, way too many indexes on some tables, and indexes that are not very useful for SELECTs. For example, IBookOrderLine.Book property - does it need an index on column Book_id? It depends - will you ever search order lines by bookId as PRIMARY criteria? Probably not.
The problem was that once you see some table over-indexed, it is really cumbersome to fight against these automatic indexes and correct the situation. So I made the change - the index on FK is automatically added only if there is a list property on the target referenced entity that lists 'this' entity. 
For example, IBookOrder.User property - if there is IUser.Orders list property, then index is added on BookOrder.User_id column - because we are going to query orders by user Id when lazy-loading list property. 
Counter example: IBookOrderLine.Book property; there is no IBook.OrderLines property on IBook, it does not make sense, there might be millions of orders/lines for a single book. So BookOrderLine.Book_id is NOT indexed automatically by VITA. 
So what you see is expected, all these FK columns are no longer indexed automatically. I suggest to go thru these old indexes, check if you really need them, and if you do, add explicit [Index] attribute on entity property
Hope this helps
